server/processors/src/test/java/org/hawkular/apm/server/processor/nodedetails/NodeDetailsDeriverTest.java

```diff
@@ -170,4 +170,67 @@ public void testInitial() throws RetryAttemptException {
         assertFalse(details.get(1).isInitial());
     }
 
+    @Test
+    public void testCalculateElapsedTimeSync() throws RetryAttemptException {
+        NodeDetailsDeriver deriver = new NodeDetailsDeriver();
+
+        Consumer consumer = new Consumer();
+        consumer.setTimestamp(1000);
+        consumer.setDuration(500);
+
+        Component component1 = new Component();
+        component1.setTimestamp(1100);
+        component1.setDuration(200);
+        consumer.getNodes().add(component1);
+
+        Component component2 = new Component();
+        component2.setTimestamp(1300);
+        component2.setDuration(200);
+        consumer.getNodes().add(component2);
+
+        assertEquals(400, deriver.calculateChildElapsedTime(consumer));
```

Comment:
The benefit of the actual time is when the child nodes are performed in its scope, to differentiate where the time is spent. The problem is when to use a simple approach for calculating this value, versus understanding that the children are async and therefore a more complex approach may need to be used.